uat/custom-components/src/main/java/com/aws/greengrass/artifacts/logGenerator.java

```diff
@@ -5,16 +5,141 @@
 
 package com.aws.greengrass.artifacts;
 
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.Logger;
+import ch.qos.logback.classic.LoggerContext;
+import ch.qos.logback.classic.encoder.PatternLayoutEncoder;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.core.rolling.RollingFileAppender;
+import ch.qos.logback.core.rolling.SizeAndTimeBasedRollingPolicy;
+import ch.qos.logback.core.util.FileSize;
 
+import java.util.concurrent.TimeUnit;
 import java.util.function.Consumer;
 
+/*
+This component rotates files with different name. Namely, the file is rotated and renamed by adding the timestamp and
+ some seq number to differ from others. The rolling policy is LogBack.TimeBasedRollingPolicy,
+ more information at https://logback.qos.ch/manual/appenders.html
+ */
 public class logGenerator implements Consumer<String[]> {
+    private static final int MAX_HISTORY = 3;
+    private static final String rotationNamePattern = "_%d{yyyy-MM-dd_HH-mm}_%i";
+    private static double random;
+    private String logFileName;
+    private String logFileExtension;
+    private double fileSize;
+    private String fileSizeUnit;
+    private double logWriteFreqSeconds;
+    private int logMsgSizeBytes;
+    private int fileSizeBytes;
+    private int totalLogNumbers;
+    // public only for unit testing
+    public String currentPath = System.getProperty("user.dir");
```

Comment:
Let's not make public attributes just for testing, you could pass it on the accept method as an arg